doc/api.md

```diff
@@ -201,6 +340,179 @@ used should be taken from those defined by
+calendar timestamp (e.g.: +002007-11-03T13:18). The middleware expects to receive the basic format in updates and return
+it in queries (and viceversa, receive the extended one in queries and return it in updates).
+
+### Timestamp Processing
+
+The IOTA processes the entity attributes looking for a `TimeInstant` attribute. If one is found, for NGSI v2, the plugin
+adds a `TimeInstant` attribute as metadata for every other attribute in the same request. With NGSI-LD, the Standard
+`observedAt` property-of-a-property is used instead.
+
+### Bidirectionality plugin (bidirectional)
+
+This plugin allows the devices with composite values an expression to update the original values in the devices when the
+composite expressions are updated in the Context Broker. This behavior is achieved through the use of subscriptions.
+
+IoTAs using this plugins should also define a notification handler to handle incoming values. This handler will be
+intercepted by the plugin, so the mapped values are included in the updated notification.
+
+When a device is provisioned with bidirectional attributes, the IoTAgent subscribes to changes in that attribute. When a
+change notification for that attribute arrives to the IoTA, it applies the transformation defined in the device
+provisioning payload to the notification, and calls the underlying notification handler with the transformed entity
+including the `value` along with any `metadata`, and in the case of an NGSI-LD bidirectional attribute a `datasetId` if
+provided.
+
+The following `attributes` section shows an example of the plugin configuration (using `IOTA_AUTOCAST=false` to avoid
+translation from geo:point to geo:json)
+
+```json
+      "attributes": [
+        {
+          "name":"location",
+          "type":"geo:point",
+          "expression": "${@latitude}, ${@longitude}",
+          "reverse": [
+            {
+              "object_id":"longitude",
+              "type": "Text",
+              "expression": "${trim(substr(@location, indexOf(@location, \",\") + 1, length(@location)))}"
+            },
+            {
+              "object_id":"latitude",
+              "type": "Text",
+              "expression": "${trim(substr(@location, 0, indexOf(@location, \",\")))}"
+            }
+          ]
+        }
+      ],
+```
+
+For each attribute that would have bidirectionality, a new field `reverse` must be configured. This field will contain
+an array of fields that will be created based on the notifications content. The expression notification can contain any
+attribute of the same entity as the bidirectional attribute; declaring them in the expressions will add them to the
+subscription payload.
+
+For each attribute in the `reverse` array, an expression must be defined to calculate its value based on the
+notification attributes. This value will be passed to the underlying protocol with the `object_id` name. Details about
+how the value is then progressed to the device are protocol-specific.
+
```

Comment:
This is the remaining plugin after latest refactor. It is included here since it is quite meaningful, configurable from API and it is the only plugin available, so makes sense including this here IMHO